tree/tree.h

```diff
@@ -15,33 +16,54 @@ class Tree{
   public:
 
     /**
-     * \brief Standard destructor.
+     * Tree class constructor.
+     * \param splitter      Splitter object (It doesn't ownership).
+     * \param criterion     Criterion object (It doesn't ownership).
+     * \param max_features    The maximum number of features used to create the tree.
+     * \param max_depth     The maximum depth of the tree.
+     * \param max_leaf_nodes  The maximum number of leaf nodes of the tree.
```

Comment:
By definition, every leaf is a node. Change this parameter to something like `max_leaves`.